wireapp/wire-android
# feat: self delete team settings [AR-1000]

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-1000" title="AR-1000" target="_blank"><img alt="Story" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10815?size=medium" />AR-1000</a>  Self-deleting message restrictions from TM
  </td></table>
  <br />
 
















----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues
- Added needed dialog to inform user when self deleting messages team settings flag has changed, and in case it has been enforced timeout team setting has been enabled, how long now the messages will be displayed.
- Added injected usecases, `ObserveSelfDeletingMessagesUseCase` and `MarkSelfDeletingMessagesChangeAsNotifiedUseCase` to observe for changes related with the self deleting messages flag and persisting the notified flag in memory to avoid recurrent dialogs.

### Dependencies (Optional)
Needs release with:

- https://github.com/wireapp/kalium/pull/1631

### Testing

#### Test Coverage (Optional)

- [x] I have added automated test to this contribution - WIP

#### How to Test

Log into a user account that belongs to a user where team settings for self deleting messages can be modified. Then modify these and check that the dialog is displayed.







----
#### PR Post Submission Checklist for internal contributors (Optional)

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
